From edf7561ce281ca2766dedb7e5d668b6ecfd9dbbc Mon Sep 17 00:00:00 2001
From: Haihao Xiang <haihao.xiang@intel.com>
Date: Mon, 13 Mar 2023 20:14:05 +0800
Subject: [PATCH 113/116] lavu/hwcontext_vaapi: relax the requirement when using
 libva2 (VAAPI 1)

With libva2, the argument for render target list to vaCreateContext() is
a hint, so we may use a dynamic frame pool.

Signed-off-by: Haihao Xiang <haihao.xiang@intel.com>
---
 doc/APIchanges              | 4 ++++
 libavutil/hwcontext_vaapi.c | 2 +-
 libavutil/hwcontext_vaapi.h | 5 +++--
 libavutil/version.h         | 2 +-
 4 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/doc/APIchanges b/doc/APIchanges
index 25cb06ddf98b..67bd8bb3564f 100644
--- a/doc/APIchanges
+++ b/doc/APIchanges
@@ -2,6 +2,10 @@ The last version increases of all libraries were on 2023-02-09
 
 API changes, most recent first:
 
+2023-09-xx - xxxxxxxxxx - lavu 58.29.100 - hwcontext_vaapi.h
+  Modify the documentation to relax the constraint for dynamic
+  frames pool.
+
 2023-09-xx - xxxxxxxxxx - lavu 58.28.100 - hwcontext_qsv.h
   Add AVQSVFramesContext.info to support dynamic frame pools
 
diff --git a/libavutil/hwcontext_vaapi.c b/libavutil/hwcontext_vaapi.c
index 53581d8df0d9..1a51df10f4d0 100644
--- a/libavutil/hwcontext_vaapi.c
+++ b/libavutil/hwcontext_vaapi.c
@@ -619,7 +619,7 @@ static int vaapi_hw_frames_init(AVHWFramesContext *hwfc)
             }
         } else {
             // This pool allows dynamic sizing, and will not be usable as a
-            // render target.
+            // render target with libva. It can be used with libva2
             avfc->nb_surfaces = 0;
             avfc->surface_ids = NULL;
         }
diff --git a/libavutil/hwcontext_vaapi.h b/libavutil/hwcontext_vaapi.h
index aea0ec926341..0e88fbe89327 100644
--- a/libavutil/hwcontext_vaapi.h
+++ b/libavutil/hwcontext_vaapi.h
@@ -26,9 +26,10 @@
  * @file
  * API-specific header for AV_HWDEVICE_TYPE_VAAPI.
  *
- * Dynamic frame pools are supported, but note that any pool used as a render
+ * Dynamic frame pools are supported. Note that any pool used as a render
  * target is required to be of fixed size in order to be be usable as an
- * argument to vaCreateContext().
+ * argument to vaCreateContext() with libva. When libva2 (VAAPI 1) is used,
+ * a pool used as a render target can be dynamic.
  *
  * For user-allocated pools, AVHWFramesContext.pool must return AVBufferRefs
  * with the data pointer set to a VASurfaceID.
diff --git a/libavutil/version.h b/libavutil/version.h
index 279e54c39441..7c0600da22f1 100644
--- a/libavutil/version.h
+++ b/libavutil/version.h
@@ -79,7 +79,7 @@
  */
 
 #define LIBAVUTIL_VERSION_MAJOR  58
-#define LIBAVUTIL_VERSION_MINOR  28
+#define LIBAVUTIL_VERSION_MINOR  29
 #define LIBAVUTIL_VERSION_MICRO 100
 
 #define LIBAVUTIL_VERSION_INT   AV_VERSION_INT(LIBAVUTIL_VERSION_MAJOR, \
-- 
2.41.0

